### DISCUSSION TITLE
How get main project info set in pyproject.toml from within project

### DISCUSSION THREAD

**Discussion by js-denton** [python-poetry/poetry#4533] [Q&A]
Hello. I have 'my-project'. For get name and version i use:
```python
import my_project
print(my_project.__version__)
print(my_project.__name__)
```
How get authors, maintainers, description, homepage?

**Comment by britodfbr** [comment]
You can use toml Package to get the information from pyproject.toml.

Toml.load("pyproject.toml")['tool']['poetry'][value]

**Reply by js-denton** [reply]
Thanks for solution, but i mean more program method without hardcode file name, parsing and after that read data.

**Reply by clintonroy** [reply]
For a local package named "poetryplay", this works for me:

```poetry run python3 -c 'from importlib.metadata import metadata; print metadata("poetryplay")["Author"]'```

**Reply by js-denton** [reply]
Thanks, i guess this is the most correct solution. Anybody know will be more simple decision?

**Comment by michael-l-a** [comment]
name:
```shell
poetry version | awk {'print $1'}
``` 

version:
```shell
poetry version | awk {'print $2'}
```

**Reply by js-denton** [reply]
Hello @m-abramovich . Thanks for your answer, but it is not the python way.
It works in a developed environment, but is not working if you distribute applications and have shown info about applications.
Most native solution is stay via `metadata`
